### PR TITLE
Refactor/the node job part 2

### PR DESCRIPTION
**Pull request by Aprogiena**
This is another step on the long journey of Node refactoring. This one includes
* rename Node to NetworkPeer (similarly some of the related classes are renamed as well)
* inject logger and date time provider into NetworkPeer
* INetworkPeerFactory and NetworkPeerFactory replaces static methods of creating NetworkPeer
* subclasses of NetworkPeer were extracted out of the class, this includes NetworkPeerConnection class 
* minor refactoring and polishing here and there

### CODE REVIEW

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 1]
```diff
@@ -134,22 +138,23 @@ public sealed class PeerConnector : IPeerConnector
...
+            this.networkPeerFactory = networkPeerFactory;
 
             this.currentParameters = this.parentParameters.Clone();
             this.currentParameters.TemplateBehaviors.Add(new PeerConnectorBehaviour(this));
             this.currentParameters.ConnectCancellation = this.nodeLifetime.ApplicationStopping;
         }
 
         /// <inheritdoc/>
-        public void AddNode(Node node)
+        public void AddNode(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Lets rename `node` to `networkPeer`

Comment (fassadlr) [reply src/Stratis.Bitcoin/P2P/PeerConnector.cs]: As well as `AddNode` to `AddNetworkPeer`

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 2]
```diff
@@ -134,22 +138,23 @@ public sealed class PeerConnector : IPeerConnector
...
         {
             Guard.NotNull(node, nameof(node));
 
             this.ConnectedPeers.Add(node);
         }
 
         /// <inheritdoc/>
-        public void RemoveNode(Node node)
+        public void RemoveNode(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: The same here, rename to `networkPeer`

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 3]
```diff
@@ -169,11 +174,11 @@ public void StartConnectAsync()
         /// <summary>Attempts to connect to a random peer.</summary>
         private Task ConnectAsync()
         {
-            Node node = null;
+            NetworkPeer node = null;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Same here, rename to `networkPeer`.

File: src/Stratis.Bitcoin/P2P/PeerConnectorBehaviour.cs [hunk 4]
```diff
@@ -27,12 +27,12 @@ protected override void DetachCore()
         }
 
         /// <inheritdoc/>
-        private void AttachedNode_StateChanged(Node node, NodeState oldState)
+        private void AttachedNode_StateChanged(NetworkPeer node, NetworkPeerState oldState)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerConnectorBehaviour.cs]: Rename node to `networkPeer`

File: src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs [hunk 5]
```diff
@@ -98,7 +103,7 @@ private Task DiscoverPeersAsync()
                 {
                     connectTokenSource.CancelAfter(TimeSpan.FromSeconds(5));
 
-                    Node node = null;
+                    NetworkPeer node = null;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs]: Same here, rename to `networkPeer`.

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/BroadcastTransactionBehavior.cs [hunk 6]
```diff
@@ -33,7 +33,7 @@ public class TransactionBroadcast
     public class BroadcastHub
     {
         internal ConcurrentDictionary<uint256, Transaction> BroadcastedTransaction;
-        internal ConcurrentDictionary<Node, Node> Nodes;
+        internal ConcurrentDictionary<NetworkPeer, NetworkPeer> Nodes;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/BroadcastTransactionBehavior.cs]: Nodes > `NetworkPeers`

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 7]
```diff
@@ -52,7 +52,7 @@ public void PushMessage(IncomingMessage message)
 
         public MessageProducer<IncomingMessage> MessageProducer { get; private set; }
 
-        private ConcurrentDictionary<Node, Node> nodes;
+        private ConcurrentDictionary<NetworkPeer, NetworkPeer> nodes;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename to `networkPeers`

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 8]
```diff
@@ -68,17 +68,17 @@ public int Count
...
+            public bool Equals(NetworkPeer nodeA, NetworkPeer nodeB)
             {
                 if ((nodeA == null) || (nodeB == null))
                     return (nodeA == null) && (nodeB == null);
 
                 return (nodeA.RemoteSocketAddress.MapToIPv6().ToString() == nodeB.RemoteSocketAddress.MapToIPv6().ToString()) && (nodeA.RemoteSocketPort == nodeB.RemoteSocketPort);
             }
 
-            public int GetHashCode(Node node)
+            public int GetHashCode(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 9]
```diff
@@ -68,17 +68,17 @@ public int Count
         /// <summary>
         /// Provides a comparer to specify how nodes are compared for equality.
         /// </summary>
-        public class NodeComparer : IEqualityComparer<Node>
+        public class NodeComparer : IEqualityComparer<NetworkPeer>
         {
-            public bool Equals(Node nodeA, Node nodeB)
+            public bool Equals(NetworkPeer nodeA, NetworkPeer nodeB)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename nodeA and nodeB.

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 10]
```diff
@@ -87,14 +87,14 @@ public int GetHashCode(Node node)
...
+        public NetworkPeerCollection()
         {
             this.MessageProducer = new MessageProducer<IncomingMessage>();
             this.bridge = new Bridge(this.MessageProducer);
-            this.nodes = new ConcurrentDictionary<Node, Node>(new NodeComparer());
+            this.nodes = new ConcurrentDictionary<NetworkPeer, NetworkPeer>(new NodeComparer());
         }
 
-        public bool Add(Node node)
+        public bool Add(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 11]
```diff
@@ -109,9 +109,9 @@ public bool Add(Node node)
             return false;
         }
 
-        public bool Remove(Node node)
+        public bool Remove(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 12]
```diff
@@ -122,49 +122,49 @@ public bool Remove(Node node)
             return false;
         }
 
-        private void OnNodeAdded(Node node)
+        private void OnNodeAdded(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 13]
```diff
@@ -122,49 +122,49 @@ public bool Remove(Node node)
...
         {
             this.Added?.Invoke(this, new NodeEventArgs(node, true));
         }
 
-        public void OnNodeRemoved(Node node)
+        public void OnNodeRemoved(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs [hunk 14]
```diff
@@ -122,49 +122,49 @@ public bool Remove(Node node)
...
-        public Node FindByEndpoint(IPEndPoint endpoint)
+        public NetworkPeer FindByEndpoint(IPEndPoint endpoint)
         {
             IPAddress ip = endpoint.Address.EnsureIPv6();
             int port = endpoint.Port;
             return this.nodes.Select(n => n.Key).FirstOrDefault(n => Match(ip, port, n));
         }
 
-        private static bool Match(IPAddress ip, int? port, Node node)
+        private static bool Match(IPAddress ip, int? port, NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerListener.cs [hunk 15]
```diff
@@ -8,25 +8,25 @@
...
-    public class NodeListener : PollMessageListener<IncomingMessage>, IDisposable
+    public class NetworkPeerListener : PollMessageListener<IncomingMessage>, IDisposable
     {
-        public Node Node { get; private set; }
+        public NetworkPeer Node { get; private set; }
         private IDisposable subscription;
         private List<Func<IncomingMessage, bool>> predicates = new List<Func<IncomingMessage, bool>>();
 
-        public NodeListener(Node node)
+        public NetworkPeerListener(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerListener.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerListener.cs [hunk 16]
```diff
@@ -8,25 +8,25 @@
 
 namespace Stratis.Bitcoin.P2P.Peer
 {
-    public class NodeListener : PollMessageListener<IncomingMessage>, IDisposable
+    public class NetworkPeerListener : PollMessageListener<IncomingMessage>, IDisposable
     {
-        public Node Node { get; private set; }
+        public NetworkPeer Node { get; private set; }
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerListener.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs [hunk 17]
```diff
@@ -69,7 +81,7 @@ public IPEndPoint ExternalEndpoint
             }
         }
 
-        public NodesCollection ConnectedNodes { get; private set; }
+        public NetworkPeerCollection ConnectedNodes { get; private set; }
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs]: Rename ConnectedNodes

File: src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs [hunk 18]
```diff
@@ -43,7 +43,7 @@ protected override void AttachCore()
             this.AttachedNode.MessageReceived += this.AttachedNode_MessageReceived;
         }
 
-        private void AttachedNode_MessageReceived(Node node, IncomingMessage message)
+        private void AttachedNode_MessageReceived(NetworkPeer node, IncomingMessage message)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs [hunk 19]
```diff
@@ -63,15 +63,15 @@ private void AttachedNode_MessageReceived(Node node, IncomingMessage message)
         //if (node.State == NodeState.Connected)
         //which is more intuitive.
         //This happens in PeerDiscovery as well where we connect and then disconnect straight after.
-        private void AttachedNode_StateChanged(Node node, NodeState previousState)
+        private void AttachedNode_StateChanged(NetworkPeer node, NetworkPeerState previousState)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 20]
```diff
@@ -38,15 +38,15 @@ public interface IPeerConnector : IDisposable
         /// This will only happen if the node successfully handshaked with another.
         /// </para>
         /// </summary>
-        void AddNode(Node node);
+        void AddNode(NetworkPeer node);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/PeerConnector.cs [hunk 21]
```diff
@@ -38,15 +38,15 @@ public interface IPeerConnector : IDisposable
...
 
         /// <summary>
         /// Removes a given node from the <see cref="NodesCollection"/>.
         /// <para>
         /// This will happen if the node state changed to "disconnecting", "failed" or "offline".
         /// </para>
         /// </summary>
-        void RemoveNode(Node node);
+        void RemoveNode(NetworkPeer node);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/PeerConnector.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/BroadcastTransactionBehavior.cs [hunk 22]
```diff
@@ -231,9 +231,9 @@ TransactionBroadcast GetTransaction(ulong pingValue, bool remove)
             return result;
         }
 
-        void AttachedNode_StateChanged(Node node, NodeState oldState)
+        void AttachedNode_StateChanged(NetworkPeer node, NetworkPeerState oldState)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/BroadcastTransactionBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/BroadcastTransactionBehavior.cs [hunk 23]
```diff
@@ -52,16 +52,16 @@ public IEnumerable<Transaction> BroadcastingTransactions
         public BroadcastHub()
         {
             this.BroadcastedTransaction = new ConcurrentDictionary<uint256, Transaction>();
-            this.Nodes = new ConcurrentDictionary<Node, Node>();
+            this.Nodes = new ConcurrentDictionary<NetworkPeer, NetworkPeer>();
             this.ManualBroadcast = false;
         }
 
-        public static BroadcastHub GetBroadcastHub(Node node)
+        public static BroadcastHub GetBroadcastHub(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/BroadcastTransactionBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehavior.cs [hunk 24]
```diff
@@ -6,8 +6,8 @@ namespace Stratis.Bitcoin.P2P.Protocol.Behaviors
 {
     public interface INodeBehavior
     {
-        Node AttachedNode { get; }
-        void Attach(Node node);
+        NetworkPeer AttachedNode { get; }
+        void Attach(NetworkPeer node);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehavior.cs [hunk 25]
```diff
@@ -29,7 +29,7 @@ protected void RegisterDisposable(IDisposable disposable)
             this.disposables.Add(disposable);
         }
 
-        public void Attach(Node node)
+        public void Attach(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehavior.cs [hunk 26]
```diff
@@ -53,9 +53,9 @@ protected void AssertNotAttached()
                 throw new InvalidOperationException("Can't modify the behavior while it is attached");
         }
 
-        private static bool Disconnected(Node node)
+        private static bool Disconnected(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehaviorsCollection.cs [hunk 27]
```diff
@@ -5,10 +5,10 @@ namespace Stratis.Bitcoin.P2P.Protocol.Behaviors
 {
     public class NodeBehaviorsCollection : ThreadSafeCollection<INodeBehavior>
     {
-        private readonly Node node;
+        private readonly NetworkPeer node;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehaviorsCollection.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehaviorsCollection.cs [hunk 28]
```diff
@@ -5,10 +5,10 @@ namespace Stratis.Bitcoin.P2P.Protocol.Behaviors
...
         private bool delayAttach;
 
-        public NodeBehaviorsCollection(Node node)
+        public NodeBehaviorsCollection(NetworkPeer node)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/NodeBehaviorsCollection.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/PingPongBehavior.cs [hunk 29]
```diff
@@ -92,13 +92,13 @@ protected override void AttachCore()
 
         private bool PingVersion()
         {
-            Node node = this.AttachedNode;
+            NetworkPeer node = this.AttachedNode;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/PingPongBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/PingPongBehavior.cs [hunk 30]
```diff
@@ -92,13 +92,13 @@ protected override void AttachCore()
...
             return (node != null) && (node.Version > NBitcoin.Protocol.ProtocolVersion.BIP0031_VERSION);
         }
 
-        void AttachedNode_StateChanged(Node node, NodeState oldState)
+        void AttachedNode_StateChanged(NetworkPeer node, NetworkPeerState oldState)
         {
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/PingPongBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Behaviors/PingPongBehavior.cs [hunk 31]
```diff
@@ -137,7 +137,7 @@ public void Probe()
 
         void PingTimeout(object ping)
         {
-            Node node = this.AttachedNode;
+            NetworkPeer node = this.AttachedNode;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Behaviors/PingPongBehavior.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Filters/ActionFilter.cs [hunk 32]
```diff
@@ -7,19 +7,19 @@ namespace Stratis.Bitcoin.P2P.Protocol.Filters
...
+            this.onSending = onSending ?? new Action<NetworkPeer, Payload, Action>((m, p, n) => n());
         }
 
         public void OnReceivingMessage(IncomingMessage message, Action next)
         {
             this.onIncoming(message, next);
         }
 
-        public void OnSendingMessage(Node node, Payload payload, Action next)
+        public void OnSendingMessage(NetworkPeer node, Payload payload, Action next)
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Filters/ActionFilter.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/Filters/NodeFilter.cs [hunk 33]
```diff
@@ -22,6 +22,6 @@ public interface INodeFilter
         /// <param name="node"></param>
         /// <param name="payload"></param>
         /// <param name="next">The rest of the pipeline</param>
-        void OnSendingMessage(Node node, Payload payload, Action next);
+        void OnSendingMessage(NetworkPeer node, Payload payload, Action next);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/Filters/NodeFilter.cs]: Rename node

File: src/Stratis.Bitcoin/P2P/Protocol/IncomingMessage.cs [hunk 34]
```diff
@@ -10,7 +10,7 @@ public class IncomingMessage
     {
         public Message Message { get; set; }
         internal Socket Socket { get; set; }
-        public Node Node { get; set; }
+        public NetworkPeer Node { get; set; }
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/P2P/Protocol/IncomingMessage.cs]: Rename node

**Comment by Aprogiena** [comment]
`mode` should be `more` sry :D

**Comment by Aprogiena** [comment]
should be ok now

**Comment by dangershony** [comment]
Ok I will look at this now

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs [hunk 35]
```diff
@@ -112,23 +138,23 @@ public NodeServer(Network network, ProtocolVersion version = ProtocolVersion.PRO
...
-            this.ConnectedNodes.Removed += Nodes_NodeRemoved;
-            this.ConnectedNodes.MessageProducer.AddMessageListener(listener);
+            this.ConnectedNetworkPeers = new NetworkPeerCollection();
+            this.ConnectedNetworkPeers.Added += Peers_PeerAdded;
+            this.ConnectedNetworkPeers.Removed += Peers_PeerRemoved;
+            this.ConnectedNetworkPeers.MessageProducer.AddMessageListener(listener);
 
             this.AllMessages = new MessageProducer<IncomingMessage>();
-            this.trace = new TraceCorrelation(NodeServerTrace.Trace, "Node server listening on " + this.LocalEndpoint);
+            this.trace = new TraceCorrelation(NodeServerTrace.Trace, "Network peer server listening on " + this.LocalEndpoint);
```
Comment (dangershony) [inline src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs]: Did we not get rid of `TraceCorrelation`?

Comment (dangershony) [reply src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs]: Ah its temporary ok fine

File: src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs [hunk 36]
```diff
@@ -219,7 +219,7 @@ public Task AnnounceBlocks(List<uint256> blockHashesToAnnounce)
                 return Task.CompletedTask;
             }
 
-            Node node = this.AttachedNode;
+            NetworkPeer node = this.AttachedPeer;
```
[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs]: i've already done thousands of those and thousands are left, so another day maybe :)

File: src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs [hunk 37]
```diff
@@ -121,7 +121,7 @@ public override void Start()
             this.connectionManager.Parameters.TemplateBehaviors.Add(new BlockPullerBehavior(this.blockPuller, this.loggerFactory));
 
             // signal to peers that this node can serve blocks
-            this.connectionManager.Parameters.Services = (this.storeSettings.Prune ? NodeServices.Nothing : NodeServices.Network) | NodeServices.NODE_WITNESS;
+            this.connectionManager.Parameters.Services = (this.storeSettings.Prune ? NetworkPeerServices.Nothing : NetworkPeerServices.Network) | NetworkPeerServices.NODE_WITNESS;
```
[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs]: PeerServices or NetworkPeerServices, seems same, or not?

File: src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs [hunk 38]
```diff
@@ -533,6 +534,10 @@ internal void AddDisposable(IDisposable group)
     public class CoreNode
     {
         private readonly NodeBuilder builder;
+
+        /// <summary>Factory for creating P2P network peer clients and servers.</summary>
```
[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin.IntegrationTests/NodeBuilder.cs]: ok if it sounds better to you, i'll fix that

File: src/Stratis.Bitcoin/Connection/ConnectionManager.cs [hunk 39]
```diff
@@ -89,27 +89,30 @@ public sealed class ConnectionManager : IConnectionManager
         /// <summary>Global application life cycle control - triggers when application shuts down.</summary>
         private readonly INodeLifetime nodeLifetime;
 
-        private readonly NodesCollection connectedNodes = new NodesCollection();
+        /// <summary>Factory for creating P2P network peer clients and servers.</summary>
+        private readonly INetworkPeerFactory networkPeerFactory;
 
-        public IReadOnlyNodesCollection ConnectedNodes
+        private readonly NetworkPeerCollection connectedNodes = new NetworkPeerCollection();
```
[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin/Connection/ConnectionManager.cs]: will do, but not my code ... i could be fixing all days long ...

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs [hunk 40]
```diff
@@ -330,11 +337,11 @@ private set
         public int RemoteSocketPort { get; private set; }
         public bool Inbound { get; private set; }
 
-        private bool reuseBuffer;
-        public NodeBehaviorsCollection Behaviors { get; private set; }
-        public NetworkAddress Peer { get; private set; }
+        public bool ReuseBuffer;
```
[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs]: yes, part of NodeConnection separation thing, will be refactored anyway, will add private setter though

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs [hunk 41]
```diff
@@ -0,0 +1,155 @@
...
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.Protocol;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.P2P.Peer
+{
+    /// <summary>
+    /// Contract for factory for creating P2P network peer clients and servers.
```
[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs]: k

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs [hunk 42]
```diff
@@ -0,0 +1,155 @@
...
+        /// <summary>
+        /// Creates a new network peer which is connected to a specified counterparty.
+        /// </summary>
+        /// <param name="network">The network to connect to.</param>
+        /// <param name="endpoint">Address of the counterparty to connect to.</param>
+        /// <param name="myVersion">Version of the protocol that the node supports.</param>
+        /// <param name="isRelay">Whether the remote peer should announce relayed transactions or not. See <see cref="VersionPayload.Relay"/> for more information.</param>
+        /// <param name="cancellation">Cancallation token that allows to interrupt establishing of the connection.</param>
+        /// <returns>Network peer connected to the specified counterparty.</returns>
+        NetworkPeer CreateConnectedNetworkPeer(Network network, string endpoint, ProtocolVersion myVersion = ProtocolVersion.PROTOCOL_VERSION, bool isRelay = true, CancellationToken cancellation = default(CancellationToken));
```
[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin/P2P/Peer/NetworkPeerFactory.cs]: this should all be rewritten later, but default cancellation is no cancellation, 
it is used in tests and tests only, we could remove it and rewrite the test, but baah for now

File: src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs (hunk 35 above)
Comment (Aprogiena) [reply src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs]: PR too big already, getting rid of this did not make it in this phase

File: src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs (hunk 37 above)
[the remark this answers is not in the document]
Comment (dangershony) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs]: NetworkPeer represents an instance of a peer while PeerServices describes the service a peer can provide.

But I am not sure on a different name yet so leave it as is.

[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs]: ok i'll update

[the remark this answers is not in the document]
Comment (Aprogiena) [reply src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs]: ok i leave it

**Pull request merged**